internal/featuregates/features.go

```diff
@@ -0,0 +1,59 @@
+package featuregates
+
+/* Design aspects of implementing feature gates
+
+- feature gate is only for experimental features
+- Keep the feature gate as simple boolean.
+- Any config specific for a feature gate should be in it’s own configMap. This aligns with our current implementation of peer-pods and image generator feature
+- When we decide to make the feature as a stable feature, it should move to kataConfig.spec
+*/
+
+const (
+	FeatureGatesConfigMapName     = "osc-feature-gates"
+	ImageBasedDeployment          = "imageBasedDeployment"
+	ImageBasedDeploymentConfigMap = "osc-feature-gate-image-deploy-config"
```

Comment:
I'd remove "feature-gate" from the name, i don't see why it needs to be mentioned
also i wonder if we can find a name for this feature which will be less confusing with podvm image creation stuff, maybe `layered-image-deployment` or something WDYT?

Reply:
It helps to know which configmaps are for what by just looking at the names. May be I can just use `fg` instead of `feature-gate`. 
I'll rename the feature gate to layered image deployment. It makes sense